### PR TITLE
PSR-11

### PR DESCRIPTION
**Pull request by Tyrsson**
See relevant comments with a search for bugo

First of all, thank you for taking the time to contribute to our portal, we really appreciate it!

Please read our [contribution guide](./CONTRIBUTING.md) first.

Please make sure to create a new branch for your PR.

If you want to add a new plugin, use title like `Add a new plugin - <PluginName>`.

If you want to add a new feature, use title like `Add feature - <Feature Title>`.

If you want to fix any open issue, use title like `Fix issue <description of the issue>`.

Remove this placeholder text and replace it with a description of what this PR is doing.

### CODE REVIEW

File: src/Sources/LightPortal/AddonHandlerAwareInterface.php
Comment (dragomano) [inline src/Sources/LightPortal/AddonHandlerAwareInterface.php:1]: It's the same as Addons/AddonManagerAwareInterface, can be removed

File: src/Sources/LightPortal/AddonHandlerAwareTrait.php
Comment (dragomano) [inline src/Sources/LightPortal/AddonHandlerAwareTrait.php:1]: It's the same as Addons/AddonManagerAwareTrait, can be removed

File: src/Sources/LightPortal/app.php [hunk 1]
```diff
@@ -37,11 +40,29 @@
...
+	//$container->get(AddonHandler::class),
+	$container->get(Actions\Block::class),
+	$container->get(Actions\BoardIndex::class),
+	$container->get(Actions\Category::class),
+	$container->get(Actions\FrontPage::class),
+	$container->get(Actions\Page::class),
+	$container->get(Actions\Tag::class)
+);
+//$int->init();
+$int();
```
Comment (dragomano) [inline src/Sources/LightPortal/app.php:66]: Got the "Too few arguments to function Bugo\LightPortal\Integration::__construct(), 0 passed in \Sources\Subs.php on line 5978 and at least 1 expected" error

File: src/Sources/LightPortal/Addons/AddonManagerFactory.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bugo\LightPortal\Addons;
+
+use Laminas\ServiceManager\Exception;
+use Laminas\ServiceManager\Factory\FactoryInterface;
```
Comment (dragomano) [inline src/Sources/LightPortal/Addons/AddonManagerFactory.php:8]: You forgot to add `use Psr\Container\ContainerInterface;`

File: src/Sources/LightPortal/AddonHandler.php [hunk 3]
```diff
@@ -75,31 +113,50 @@ public function run(string $hook = 'init', array $vars = [], array $plugins = []
...
+			 */
 			if (method_exists($class, $hook)) {
 				$hook === 'init' && in_array($addon, Utils::$context['lp_enabled_plugins'])
 					? $class->init()
 					: $class->$hook(...$vars);
 			}
 		}
-
+		// no idea why we have a random call to $this->minify() here, I mean really....
 		$this->minify();
```
Comment (dragomano) [inline src/Sources/LightPortal/AddonHandler.php:160]: It's a single common call after adding assets by each plugin (see `loadAssets`)

**Review by dragomano — CHANGES_REQUESTED** [review]

File: src/Sources/LightPortal/Helper.php [hunk 4]
```diff
@@ -56,7 +56,8 @@ public function session(?string $key = null): Session
 
 	public function hook(string $hook, array $vars = [], array $plugins = []): void
 	{
-		AddonHandler::getInstance()->run($hook, $vars, $plugins);
+		//AddonHandler::getInstance()->run($hook, $vars, $plugins);
```
Comment (dragomano) [inline src/Sources/LightPortal/Helper.php:59]: It's also used in these files:
- Themes/LightPortal/ViewPage.template.php
- Sources/LightPortal/Utils/Content.php
- Sources/LightPortal/Lists/PluginList.php

File: src/Sources/LightPortal/AddonHandlerAwareInterface.php
Comment (Tyrsson) [reply src/Sources/LightPortal/AddonHandlerAwareInterface.php:1]: They are not the same. For the moment,  neither can be removed.

AddonManagerAwareInterface is the contract for accepting a AddonManager instance.

AddonHandlerAwareInterface is the contract for accepting a AddonHandler instance.

Not the same.

File: src/Sources/LightPortal/Addons/ConfigProvider.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    /**
+     * Default configuration to seed the AddonManager
+     * @return array
+     */
+    public function getAddonsConfig(): array
+    {
+        return [
+            'initializers' => [],
+            'aliases'      => [
+				'userInfo' => Addons\UserInfo\UserInfo::class,
```
Comment (dragomano) [inline src/Sources/LightPortal/Addons/ConfigProvider.php:45]: I need snake_case here: `'user_info' => Addons\UserInfo\UserInfo::class`

File: src/Sources/LightPortal/Env.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+defined('LP_NAME') || define('LP_NAME', 'Light Portal');
+defined('LP_VERSION') || define('LP_VERSION', '2.6.3');
+defined('LP_PLUGIN_LIST') || define('LP_PLUGIN_LIST', 'https://d8d75ea98b25aa12.mokky.dev/addons');
+defined('LP_ADDON_URL') || define('LP_ADDON_URL', Config::$boardurl . '/Sources/LightPortal/Addons');
+defined('LP_ADDON_DIR') || define('LP_ADDON_DIR', __DIR__ . '/Addons');
+defined('LP_CACHE_TIME') || define('LP_CACHE_TIME', (int) (Config::$modSettings['lp_cache_update_interval'] ?? 72000));
+defined('LP_ACTION') || define('LP_ACTION', Config::$modSettings['lp_portal_action'] ?? 'portal');
+defined('LP_PAGE_PARAM') || define('LP_PAGE_PARAM', Config::$modSettings['lp_page_param'] ?? 'page');
+defined('LP_BASE_URL') || define('LP_BASE_URL', Config::$scripturl . '?action=' . LP_ACTION);
```
Comment (dragomano) [inline src/Sources/LightPortal/Env.php:17]: Unfortunately, in this place `$scripturl` is not defined yet

File: src/Sources/LightPortal/entry.php
Comment (dragomano) [inline src/Sources/LightPortal/entry.php:1]: Is this a new entry point? Then you should replace `app.php` with `entry.php` in `src/package-info.xml`

File: src/Sources/LightPortal/Addons/ConfigProvider.php (hunk 5 above)
Comment (Tyrsson) [reply src/Sources/LightPortal/Addons/ConfigProvider.php:45]: Well then all I have to do is create the service Alias as user_info ;)

File: src/Sources/LightPortal/Env.php (hunk 6 above)
Comment (Tyrsson) [reply src/Sources/LightPortal/Env.php:17]: Honestly, the intention was not necessarily for it to remain this way. It's just my crazy way of isolating things. 
With a future goal of a standalone script all of the SMF integration needs isolated so as to run only in that context. Just testing what's doin what etc. some times the best way to learn something is to break it.

File: src/Sources/LightPortal/entry.php
Comment (Tyrsson) [reply src/Sources/LightPortal/entry.php:1]: Again, this was me starting from a blank file and putting it back together so I had a better feel for initialization

**Pull request closed without merging**